etl/extractor/platform_impl.go

```diff
@@ -211,34 +212,105 @@ func closeStream(ctx context.Context, stream exporter.RecordExporter_ExportClien
 	}
 }
 
-func (m *PlatformExtractor) Stop(ctx context.Context) error {
-	m.startStopMutex.Lock()
-	defer m.startStopMutex.Unlock()
-	if m.hasStarted {
+func (e *PlatformExtractor) Stop(ctx context.Context) error {
+	e.cancel()
+	e.startStopMutex.Lock()
+	defer e.startStopMutex.Unlock()
+	if e.hasStarted {
 		belogger.FromContext(ctx).Info("Stopping platform extractor...")
-		m.stopSignal <- true
-		m.hasStarted = false
+		e.stopSignal <- true
+		e.hasStarted = false
 	}
 	return nil
 }
 
-func (m *PlatformExtractor) needStop() bool {
+func (e *PlatformExtractor) needStop() bool {
 	select {
-	case <-m.stopSignal:
+	case <-e.stopSignal:
 		return true
 	default:
 		// continue
 	}
 	return false
 }
 
-func (m *PlatformExtractor) Start(ctx context.Context) error {
-	m.startStopMutex.Lock()
-	defer m.startStopMutex.Unlock()
-	if !m.hasStarted {
+func (e *PlatformExtractor) Start(ctx context.Context) error {
+	e.startStopMutex.Lock()
+	defer e.startStopMutex.Unlock()
+	if !e.hasStarted {
 		belogger.FromContext(ctx).Info("Starting platform extractor...")
-		m.getJetDropsContinuously(ctx)
-		m.hasStarted = true
+		// e.getJetDropsContinuously(ctx)
+		e.hasStarted = true
+		ctx, e.cancel = context.WithCancel(ctx)
+		go e.retrievePulses(ctx)
 	}
 	return nil
 }
+
+func (e *PlatformExtractor) retrievePulses(ctx context.Context) {
+	pu := new(exporter.FullPulse)
+	var err error
+	logger := belogger.FromContext(ctx)
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		default:
+		}
+		time.Sleep(5 * time.Second)
```

Comment:
is it possible to use the config variable here instead of 5?

Reply:
yes